velox/exec/tests/AggregationTest.cpp

```diff
@@ -3038,4 +2990,130 @@ TEST_F(AggregationTest, memoryPoolAllocationAtInit) {
 
   assertQuery(plan, "SELECT c0, cast(NULL as INT) FROM tmp GROUP BY c0");
 }
+
+DEBUG_ONLY_TEST_F(AggregationTest, reclaimEmptyInput) {
+  constexpr int64_t kMaxBytes = 1LL << 30; // 1GB
+  auto rowType = ROW({"c0", "c1", "c2"}, {INTEGER(), INTEGER(), VARCHAR()});
+  const int32_t numBatches = 5;
+  auto batches = makeVectors(rowType, numBatches, 100);
+
+  std::atomic_bool injectReclaimOnce{true};
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::exec::Values::getOutput",
+      std::function<void(const exec::Values*)>([&](const exec::Values* values) {
+        if (!injectReclaimOnce.exchange(false)) {
+          return;
+        }
+        auto* driver = values->testingOperatorCtx()->driver();
+        auto task = values->testingOperatorCtx()->task();
+        {
+          MemoryReclaimer::Stats stats;
+          SuspendedSection suspendedSection(driver);
+          task->pool()->reclaim(kMaxBytes, stats);
+          ASSERT_EQ(stats.numNonReclaimableAttempts, 0);
+        }
+        static_cast<memory::MemoryPoolImpl*>(task->pool())
+            ->testingSetCapacity(kMaxBytes);
+      }));
+
+  auto tempDirectory = exec::test::TempDirectoryPath::create();
+  auto queryCtx = std::make_shared<core::QueryCtx>(executor_.get());
+  queryCtx->testingOverrideMemoryPool(
+      memory::defaultMemoryManager().addRootPool(
+          queryCtx->queryId(), kMaxBytes, memory::MemoryReclaimer::create()));
+  core::PlanNodeId aggNodeId;
+  auto task =
+      AssertQueryBuilder(
+          PlanBuilder()
+              .values(batches)
+              // Set fake filter to ensure empty input to aggregation operator.
+              .filter("c0 != c0")
+              .singleAggregation({"c0", "c1"}, {"array_agg(c2)"})
+              .capturePlanNodeId(aggNodeId)
+              .planNode(),
+          duckDbQueryRunner_)
+          .spillDirectory(tempDirectory->path)
+          .queryCtx(queryCtx)
+          .config(QueryConfig::kSpillEnabled, "true")
+          .config(QueryConfig::kAggregationSpillEnabled, "true")
+          .assertEmptyResults();
+  auto& planStats = exec::toPlanStats(task->taskStats()).at(aggNodeId);
+  ASSERT_EQ(planStats.spilledBytes, 0);
+  OperatorTestBase::deleteTaskAndCheckSpillDirectory(task);
+}
+
+DEBUG_ONLY_TEST_F(AggregationTest, reclaimWithEmptyOutput) {
```

Comment:
consistency: reclaimEmptyInput vs. reclaimWithEmptyOutput

maybe rename to reclaimEmptyOutput